stan/math/prim/fun/append_row.hpp

```diff
@@ -161,21 +55,21 @@ inline Eigen::Matrix<T, Eigen::Dynamic, 1> append_row(
  *
  * This function applies to (scalar, vector) and returns a vector.
  *
- * @tparam T1 type of the scalar
- * @tparam T2 type of elements in the vector
- * @tparam R number of rows, can be Eigen::Dynamic
- * @tparam C number of columns, can be Eigen::Dynamic
+ * @tparam Scal type of the scalar
+ * @tparam ColVec type of the vector
  *
  * @param A scalar.
  * @param B vector.
  * @return Result of stacking the scalar on top of the vector.
  */
-template <typename T1, typename T2, int R, int C>
-inline Eigen::Matrix<return_type_t<T1, T2>, Eigen::Dynamic, 1> append_row(
-    const T1& A, const Eigen::Matrix<T2, R, C>& B) {
+template <typename Scal, typename ColVec,
+          require_stan_scalar_t<Scal>* = nullptr,
+          require_t<is_eigen_col_vector<ColVec>>* = nullptr>
+inline Eigen::Matrix<return_type_t<Scal, ColVec>, Eigen::Dynamic, 1> append_row(
+    const Scal& A, const ColVec& B) {
   using Eigen::Dynamic;
   using Eigen::Matrix;
-  using return_type = return_type_t<T1, T2>;
+  using return_type = return_type_t<Scal, ColVec>;
```

Comment:
This should be ```T_return``` for consistency with the above functions

Reply:
fixed